petab/v2/lint.py

```diff
@@ -484,11 +509,14 @@ def run(self, problem: Problem) -> ValidationIssue | None:
             c.id for c in problem.condition_table.conditions
         }
         for experiment in problem.experiment_table.experiments:
-            missing_conditions = {
-                period.condition_id
-                for period in experiment.periods
-                if period.condition_id is not None
-            } - available_conditions
+            missing_conditions = (
+                set(
+                    chain.from_iterable(
+                        period.condition_ids for period in experiment.periods
+                    )
+                )
+                - available_conditions
+            )
```

Comment:
Same thing for here and below, is it useful to add this to the classes? Looks like a lot of additional interface though so fine as is
```python
problem.experiment_table.get_condition_ids()
problem.experiment_table.get_experiment(experiment_id).get_condition_ids()
```

Reply:
I am not sure yet. I think, usually one only needs the conditions for a single period, not for the full experiment at once.